samples/net/openthread/coap_client/prj.conf

```diff
@@ -0,0 +1,88 @@
+# Generic networking options
+CONFIG_NETWORKING=y
+CONFIG_NET_UDP=y
+CONFIG_NET_IPV6=y
+CONFIG_NET_SOCKETS=y
+CONFIG_NET_SOCKETS_POSIX_NAMES=y
+CONFIG_NET_SOCKETS_POLL_MAX=4
+CONFIG_NET_CONNECTION_MANAGER=y
+
+# Enable CoAP protocol
+CONFIG_COAP=y
+CONFIG_HEAP_MEM_POOL_SIZE=2048
+
+# Kernel options
+CONFIG_MAIN_STACK_SIZE=2048
+CONFIG_ENTROPY_GENERATOR=y
+CONFIG_TEST_RANDOM_GENERATOR=n
+CONFIG_INIT_STACKS=y
+
+# Logging
+CONFIG_NET_LOG=y
+CONFIG_LOG=y
+CONFIG_NET_STATISTICS=y
+CONFIG_PRINTK=y
+
+# Network buffers
+CONFIG_NET_PKT_RX_COUNT=16
+CONFIG_NET_PKT_TX_COUNT=16
+CONFIG_NET_BUF_RX_COUNT=80
+CONFIG_NET_BUF_TX_COUNT=80
+CONFIG_NET_CONTEXT_NET_PKT_POOL=y
+
+# IP address options
+CONFIG_NET_IF_UNICAST_IPV6_ADDR_COUNT=3
+CONFIG_NET_IF_MCAST_IPV6_ADDR_COUNT=4
+CONFIG_NET_MAX_CONTEXTS=10
+
+# Network shell
+CONFIG_NET_SHELL=y
+
+CONFIG_CPLUSPLUS=y
+
+CONFIG_BT=n
+
+CONFIG_NET_CONFIG_SETTINGS=y
+CONFIG_NET_TCP=n
+CONFIG_NET_IPV4=n
```

Comment:
PLease verify that but I think at least CONFIG_NET_IPV4 is not needed.  According to documentation https://docs.zephyrproject.org/2.1.0/reference/kconfig/CONFIG_NET_IPV4.html?highlight=config_net_ipv4#cmdoption-arg-config-net-ipv4 it defaults to n. Same goes to CONFIG_NET_TCP, CONFIG_NET_CONFIG_NEED_IPV4. People will copy this config file when writing their own apps so we should keep minimal set of settings here.

Reply:
fixed